packages/framework/src/Model/Product/Parameter/ParameterFacade.php

```diff
@@ -234,4 +238,53 @@ public function getParameterValueById(int $parameterValueId): ParameterValue
     {
         return $this->parameterRepository->getParameterValueById($parameterValueId);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter $parameter
+     * @return \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterValue[]
+     */
+    public function getParameterValuesByParameter(Parameter $parameter): array
+    {
+        return $this->parameterRepository->getParameterValuesByParameter($parameter);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter $parameter
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterValueConversionData[] $parameterValuesConversionDataIndexedByParameterValueId
+     */
+    public function updateParameterValuesByConversion(
+        Parameter $parameter,
+        array $parameterValuesConversionDataIndexedByParameterValueId,
+    ): void {
+        foreach ($parameterValuesConversionDataIndexedByParameterValueId as $parameterValueId => $parameterValueConversionData) {
+            $parameterValue = $this->parameterRepository->getParameterValueById($parameterValueId);
+            $parameterValueData = $this->parameterValueDataFactory->createFromParameterValue($parameterValue);
+
+            $parameterValueData->uuid = null;
+            $parameterValueData->text = $parameterValueConversionData->newValueText;
+            $parameterValueData->numericValue = $parameterValueConversionData->newValueText;
+
+            $newParameterValue = $this->parameterValueFactory->create($parameterValueData);
```

Comment:
why do not we just update the existing values?

`updateParameterValueInProductsByConversion` would be unnecessary then, right?

Reply:
Because they might be associated with other types of parameters